src/lib/y2storage/boot_requirements_strategies/analyzer.rb

```diff
@@ -0,0 +1,199 @@
+      # @param planned_devices [Array<Planned::Device>] devices that are already planned to be
+      #   added to the starting devicegraph.
+      # @param boot_disk_name  [String, nil] device name of the disk that the system will try to
+      #   boot first. Only useful in some scenarios like legacy boot.
+      def initialize(devicegraph, planned_devices, boot_disk_name)
+        @devicegraph = devicegraph
+        @planned_devices = planned_devices
+        @boot_disk_name = boot_disk_name
+
+        @root_planned_dev = planned_devices.find do |dev|
+          dev.respond_to?(:mount_point) && dev.mount_point == "/"
+        end
+        @root_filesystem = devicegraph.filesystems.find { |fs| fs.mountpoint == "/" }
+      end
+
+      # Disk in which the system will look for the bootloader.
+      #
+      # This is relevant only in some strategies (mainly legacy boot).
+      #
+      # There is no way to query the system for such value, so this method
+      # relies on #boot_disk_name. If that information is not available, the
+      # (first) disk hosting the root filesystem is considered to be the boot
+      # disk.
+      #
+      # If "/" is still not present in the devicegraph or the list of planned
+      # devices, the first disk of the system is used as fallback.
+      #
+      # @return [Disk]
+      def boot_disk
+        return @boot_disk if @boot_disk
+
+        if boot_disk_name
+          @boot_disk = devicegraph.disks.find { |d| d.name == boot_disk_name }
+        end
+
+        @boot_disk ||= boot_disk_from_planned_dev
+        @boot_disk ||= boot_disk_from_devicegraph
+        @boot_disk ||= devicegraph.disks.first
+
+        @boot_disk
+      end
+
+      # Whether the root (/) filesystem is going to be in a LVM logical volume
+      #
+      # @return [Boolean] true if the root filesystem is going to be in a LVM
+      #   logical volume.
+      def root_in_lvm?
+        if root_planned_dev
+          root_planned_dev.is_a?(Planned::LvmLv)
+        elsif root_filesystem
+          root_filesystem.plain_blk_devices.any? { |dev| dev.is?(:lvm_lv) }
+        else
+          false
+        end
+      end
+
+      # Whether the root (/) filesystem is going to be in an encrypted device
+      #
+      # @return [Boolean] true if the root filesystem is going to be in an
+      #   encrypted device.
+      def encrypted_root?
+        if root_planned_dev
+          root_planned_dev.respond_to?(:encrypt?) && root_planned_dev.encrypt?
+        elsif root_filesystem
+          root_filesystem.plain_blk_devices.any? { |d| d.respond_to?(:encrypted?) && d.encrypted? }
+        else
+          false
+        end
+      end
+
+      # Whether the root (/) filesystem is going to be Btrfs
+      #
+      # @return [Boolean] true if the root filesystem is going to be Btrfs
+      def btrfs_root?
+        if root_planned_dev
+          root_planned_dev.respond_to?(:filesystem_type) && root_planned_dev.filesystem_type.is?(:btrfs)
```

Comment:
can it happen that device do not respond to `filesystem_type` ? actually from my POV, problem can be that `filesystem_type` is nil and then `root_planned_dev.filesystem_type.is?` crashes with access nil.